lib/livebook/settings.ex

```diff
@@ -255,5 +255,30 @@ defmodule Livebook.Settings do
   Returns the home directory in the default file system.
   """
   @spec default_file_system_home() :: FileSystem.File.t()
-  def default_file_system_home(), do: FileSystem.File.new(default_file_system())
+  def default_file_system_home() do
+    if file = get_default_dir() do
+      file
+    else
+      FileSystem.File.new(default_file_system())
+    end
+  end
+
+  @doc """
+  Sets default directory.
+  """
+  @spec set_default_dir(FileSystem.File.t()) :: :ok
+  def set_default_dir(file) do
+    Storage.insert(:settings, "global", default_dir: file)
+  end
+
+  @doc """
+  Gets default directory.
+  """
+  @spec get_default_dir() :: FileSystem.File.t() | nil
```

Comment:
We should never return `nil.` The idea is to remove `set_default_file_system`, instead we only have `set_default_dir` (which is default file system + default path within that file system).

If there is no default dir set, we should return `FileSystem.File.new(Livebook.Config.local_file_system())`.

Reply:
So remove `Settings.set_default_file_system/1` function?
Then also delete `Settings.default_file_system/0`, `Settings.default_file_system_id/0`, `Settings.remove_file_system/1` and `Settings.default_file_system_home/0`. right?

Reply:
Yeah, except `Settings.remove_file_system/1`, it is for removing any file system.

Wherever we use `Settings.default_file_system_home` we should now use `Settings.default_dir`.

Reply:
(Let's rename `get_default_dir` to `default_dir`).

Reply:
Now, remove `Settings.default_file_system_home` and rename `Settings.get_default_dir` to `Settings.default_dir`.
So, How can I remove `Settings.default_file_system_home` and replace `Settings. set_default_dir`?

Reply:
Looks good, I will add a migration to convert `:default_file_system_id` to `:default_dir` and remove the relevant functions, thanks :)